tensorflow-framework/src/test/java/org/tensorflow/framework/regularizers/L1L2Test.java

```diff
@@ -0,0 +1,120 @@
+package org.tensorflow.framework.regularizers;
+
+import org.junit.jupiter.api.Test;
+import org.tensorflow.Operand;
+import org.tensorflow.framework.utils.TestSession;
+import org.tensorflow.op.Ops;
+import org.tensorflow.types.TFloat32;
+import org.tensorflow.types.TFloat64;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class L1L2Test extends CommonTest {
+  private final TestSession.Mode[] tfModes = {TestSession.Mode.EAGER, TestSession.Mode.GRAPH};
+
+  @Test
+  public void testCreate() {
+    for (TestSession.Mode tfMode : tfModes)
+      try (TestSession session = TestSession.createTestSession(tfMode)) {
+        Ops tf = session.getTF();
+        L1L2<TFloat32> instance = new L1L2<>(tf, 0.2f, 0.3f, TFloat32.class);
+        assertEquals(0.2f, instance.getL1());
+        assertEquals(0.3f, instance.getL2());
+
+        instance = new L1L2<>(tf, null, null, TFloat32.class);
+        assertEquals(0.f, instance.getL1());
+        assertEquals(0.f, instance.getL2());
+
+        instance = new L1L2<>(tf, 0.5f, null, TFloat32.class);
+        assertEquals(0.5f, instance.getL1());
+        assertEquals(0.f, instance.getL2());
+
+        instance = new L1L2<>(tf, null, 0.5f, TFloat32.class);
+        assertEquals(0.f, instance.getL1());
+        assertEquals(0.5f, instance.getL2());
+      }
+  }
+
+  @Test
+  public void testCallDefaultsConstant() {
+    for (TestSession.Mode tfMode : tfModes)
+      try (TestSession session = TestSession.createTestSession(tfMode)) {
+        Ops tf = session.getTF();
+        L1L2<TFloat32> instance = new L1L2<>(tf, TFloat32.class);
+        Operand<TFloat32> result = instance.call(tf.constant(555f));
+        session.evaluate(0f, result);
+      }
+  }
+
+  @Test
+  public void testCallL1L20() {
+    for (TestSession.Mode tfMode : tfModes)
+      try (TestSession session = TestSession.createTestSession(tfMode)) {
+        Ops tf = session.getTF();
+        L1L2<TFloat32> instance = new L1L2<>(tf, TFloat32.class);
+        Operand<TFloat32> weights =
+            tf.constant(new float[][] {{1.0f, 0.9f, 0.8f}, {1.2f, 0.7f, 1.1f}});
+        Operand<TFloat32> result = instance.call(weights);
+        session.evaluate(0, result);
+      }
+  }
+
+  @Test
+  public void testCallL1L2TFloat32() {
+    for (TestSession.Mode tfMode : tfModes)
+      try (TestSession session = TestSession.createTestSession(tfMode)) {
+        Ops tf = session.getTF();
+        L1L2<TFloat32> instance = new L1L2<>(tf, 0.01f, 0.02f, TFloat32.class);
+        float[][] w = {{1.0f, 0.9f, 0.8f}, {1.2f, 0.7f, 1.1f}};
+        Operand<TFloat32> weights = tf.constant(w);
+        Operand<TFloat32> result = instance.call(weights);
+        float expected = regularizeL1L2(w, 0.01f, 0.02f);
+        session.setEpsilon(.09f);
```

Comment:
There seems to be a wide tolerance needed when calculating in Java vs calculation in TF. Not sure why, but seems to happen a lot.

Reply:
That seems concerning to me! For most (reasonably stable) computations, for the first N significant digits, there's only one right answer. Or am I missing something?

Reply:
I am not sure, but I have seen a wide difference in the number of significant digits in other test cases when I use Java to determine the expected result.

Reply:
👍 Ok, seems this is a broader issue, not part of this PR. I'll raise an issue.

Reply:
#219

Reply:
I do notice that if I use `NumPy`  to do the test calculation beforehand, rather than use Java to do the calculation at runtime, things are closer to TF. I am not sure if my calculations don't exactly match `NumPy` behavior or not. I have found subtle differences between Java and Python behavior in other areas when it comes to Math which may be part of the problem. `Negative 0` being one of them.